internal/compiler/lookup.rs

```diff
@@ -461,24 +471,44 @@ impl LookupObject for ElementRc {
     }
 }
 
+pub enum StyleMetricsPropertyUse {
+    Acceptable,
+    Deprecated(String),
+    Unacceptable,
+}
+
+/// Returns Some() if name is a deprecated style metrics property.
+/// Returns None if the property is acceptable for regular use.
+/// Returns Err(()) if the property should not be accessible.
 pub fn check_deprecated_stylemetrics(
     elem: &ElementRc,
     ctx: &LookupCtx<'_>,
     name: &SmolStr,
-) -> Option<String> {
+) -> StyleMetricsPropertyUse {
     let borrow = elem.borrow();
+
+    let is_style_metrics_prop = matches!(
+        borrow.enclosing_component.upgrade().unwrap().id.as_str(),
+        "StyleMetrics" | "NativeStyleMetrics"
+    );
+
     (!ctx.type_register.expose_internal_types
-        && matches!(
-            borrow.enclosing_component.upgrade().unwrap().id.as_str(),
-            "StyleMetrics" | "NativeStyleMetrics"
-        )
+        && is_style_metrics_prop
         && borrow
             .debug
             .first()
             .and_then(|x| x.node.source_file())
             .map_or(true, |x| x.path().starts_with("builtin:"))
         && !name.starts_with("layout-"))
     .then(|| format!("Palette.{name}"))
+    .map_or(StyleMetricsPropertyUse::Acceptable, |msg| {
+        if !ctx.type_register.expose_internal_types && is_style_metrics_prop && name == "style-name"
```

Comment:
```suggestion
        if is_style_metrics_prop && name == "style-name"
```

I think that's a dead condition since it is checked earlier already.

Reply:
Good point, otherwise we wouldn’t be in that branch. Thanks for spotting:)